godtools/Managers/DownloadResources/DownloadedResourceManager.swift

```diff
@@ -52,79 +52,113 @@ class DownloadedResourceManager: GTDataManager {
-                cachedResource.addToTranslations(cachedTranslation)
+                cachedResource!.code = remoteResource.abbreviation!
+                cachedResource!.name = remoteResource.name!
+                cachedResource!.copyrightDescription = remoteResource.copyrightDescription
+                cachedResource!.bannerRemoteId = remoteResource.bannerId
+                cachedResource!.totalViews = remoteResource.totalViews!.int32Value
                 
-                TranslationsManager.shared.purgeTranslationsOlderThan(cachedTranslation, saving: false)
-            }
-            
-            let remotePages = remoteResource.pages!
-            for remotePageGeneric in remotePages {
-                let remotePage = remotePageGeneric as! PageResource
+                for remoteAttachment in (remoteResource.attachments!) {
+                    let remoteAttachment = remoteAttachment as! AttachmentResource
+                    var cachedAttachment = findEntity(Attachment.self,
+                                                      byAttribute: "remoteId",
+                                                      withValue: remoteAttachment.id!)
+                    
+                    if cachedAttachment == nil {
+                        cachedAttachment = Attachment()
+                        cachedAttachment!.remoteId = remoteAttachment.id!
+                        realm.add(cachedAttachment!)
+                    }
+                    
+                    cachedAttachment!.sha = remoteAttachment.sha256
+                    cachedAttachment!.resource = cachedResource
+                }
                 
-                let cachedPage = findFirstOrCreateEntity(PageFile.self,
-                                                         byAttribute: "remoteId",
-                                                         withValue: remotePage.id!)
+                if cachedResource!.bannerRemoteId != nil {
+                    _ = BannerManager.shared.downloadFor(cachedResource!)
+                }
                 
-                cachedPage.filename = remotePage.filename
-                cachedPage.resource = cachedResource
+                let remoteTranslations = remoteResource.latestTranslations!
+                for remoteTranslationGeneric in remoteTranslations {
+                    let remoteTranslation = remoteTranslationGeneric as! TranslationResource
+                    let languageId = remoteTranslation.language?.id ?? "-1"
+                    let resourceId = remoteResource.id ?? "-1"
+                    let version = remoteTranslation.version!.int16Value
+                    
+                    if !translationShouldBeSaved(languageId: languageId, resourceId: resourceId, version: version) {
+                        continue;
+                    }
+                    
+                    var cachedTranslation = findEntity(Translation.self,byAttribute: "remoteId",withValue: remoteTranslation.id!)
```

Comment:
Maybe we can make methods that use the decorator pattern, like: `findEntityByRemoteId`, what do you think?